### ISSUE TITLE
Separate SID for 2.4 and 5 Mhz channels?

### ISSUE THREAD

**Issue by RobinE** [latonita/tenda-reverse#1]
Congratulations on this exploration.
It would be interesting to have a separate SID for the two WiFi channels so that a client (Amazon Echo) would not automatically switch between the two WiFi channels as signal strength varies.  The Amazon Echo experiences pauses in audio streaming and this seems to be caused by this switching with Tenda mesh routers.
Have you seen any way to have separate SIDs or to switch off one of the WiFi channels?

**Comment by crees** [comment]
I think that would need doing on all of the nodes, so I'd be nervous about messing with them with the cfm utility- you'd risk having only one node having 2.4GHz at a different SSID.

Are you certain the device is doing this?  I think it more likely it's switching between nodes.  Have you tried turning off fast roaming?

**Comment by latonita** [comment]
yeah, thats risky. Another thought - Redis is used to synchronize/propogate configuration changes, maybe one can write the needed change to redis and it will copy parameter to other nodes

**Comment by RobinE** [comment]
I've dealt with the pauses by using the big Echo locked to a separate 2.4 GHz SID on my ISP provided modem/router,  then a different SID on the Tenda mesh for everything else on the site.  The big Echo doesn't go walk-about!  A different Echo Dot doesn't have the problem of pauses with the Tenda mesh, but that could simply be due to positioning.  An Amazon Fire Stick and a Google Chromecast have never showed any problem with the Tenda mesh, but all could be due to fortuitous positioning. 
It's only an ad-hoc solution, but it solved the problem, and in every other respect the Tenda meshes have worked splendidly.

**Comment by scottty27** [comment]
Just to add to this - I have successfully changed the 2.4 GHz SSID to a different name via the cfm utility - I had to change it on every node and it works fine and persists after reboot. I even changed the SSIDs of the 5 GHz network for each node so I can choose which node I connect to - it seems devices don't always like to pick the closest node. The units still connect to each other (presumably via a hidden SSID) just fine so mesh functionality isn't affected.

So basically I have one SSID as a 'mesh' running on 2.4 GHz for general use and then faster 5 GHz ones where I can choose which node I connect to. 

One thing I noticed though is that the nodes try to manage the 5 GHz network but hiding the SSID periodically to try and 'steer' devices to other nodes, obviously if the SSIDs are different this won't work, but it explains why I could be right next to a node and it would connect to one further away. 

I'm not sure how to stop this behaviour but running "iwpriv wlan0 td_ssid_hide 0" unhides the SSID again, until the Tenda scripts hide it. I therefore write a basic bash script stored in the /tmp directory that runs every 10 seconds to unhide the SSID. This works fine but obviously has to be set up again via telnet after every reboot, and I've noticed the Tenda units get very unstable after around 7-8 days, so I have to set this up again every week. A bit of a pain, but definitely worth it.

Basically I log in via telnet and run the following:

cd /tmp
echo "while true" >> "5g_wifienable.sh"
echo "do" >> "5g_wifienable.sh"
echo "    iwpriv wlan0 td_ssid_hide 0" >> "5g_wifienable.sh"
echo "    sleep 10" >> "5g_wifienable.sh"
echo "done" >> "5g_wifienable.sh"
chmod +x 5g_wifienable.sh
sh 5g_wifienable.sh &


If anyone has a better idea on how to stop the 5 GHz SSID auto hiding, please share!

**Comment by gingerbeardman** [comment]
@scottty27 how is it when you walk between meshes? device like iPhone doesn't interrupt as it changes between access points with different SSID?

**Comment by scottty27** [comment]
I usually have my iPhone on the 2.4 GHz band which I have as the same SSID for all nodes. 

For 5 GHz, if you have all the different 5 GHz networks saved and set to auto join on the iPhone it usually does switch between and if not, a quick tap of wifi off and on in the control centre usually connects to the strongest 5G node. Failing that I just choose which node (network) manually - not exactly ideal but in my opinion better than the stock option where it would quite frequently get stuck connected to a node with a weak connection.

I was nearly ready to return my mesh system for a refund as with the lack of control out of the box it would so frequently get stuck on a node that was far away and would struggle to even get 10 Mbps speeds - now with the way I have it set up I can just connect to my 2.4 GHz SSID if I want easy roaming, but can easily switch to a specific 5 GHz SSID where it always maxes out my internet speeds (150 Mbps fibre). The hardware is definitely capable but the firmware in its default state is lacking.

Many thanks to @latonita for figuring out how to telnet in to these, as they would be no good for me without it!

**Comment by gingerbeardman** [comment]
Thanks for the insight, it's good to have this ability

**Comment by RobinE** [comment]
Thanks guys for all your work on this ancient issue.  I will dig out my set of 4 [neat looking] Tenda's out of the cupboard (never throw anything away) and experiment.  I'm currently using 3 BT Home Hub mesh WiFi discs which allow separate SSIDs on the 2.4 and 5 GHz channels, but not getting the "whole home" WiFi I would like with lots of Echo dots and IOT gizmos.  As I understand it, the solution is to telnet in, fiddle around, and then use the built-in cfm configuration utility.

**Comment by latonita** [comment]
@scottty27 @RobinE  there is still room for research - to find a way to inject a script or have ssh or telnet up and running, so any following changes can be done w/o physical access... I would love to continue the research, but have almost no access to these cubes at the moment.
